platform-emf/src/main/java/com/softicar/platform/emf/page/EmfPageBadge.java

```diff
@@ -0,0 +1,141 @@
+package com.softicar.platform.emf.page;
+
+import com.softicar.platform.dom.refresh.bus.IDomRefreshBusEvent;
+import com.softicar.platform.dom.refresh.bus.IDomRefreshBusListener;
+import com.softicar.platform.emf.validation.result.EmfDiagnosticLevel;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * Defines a badge on a {@link IEmfPage} navigation link.
+ *
+ * @author Oliver Richers
+ */
+public class EmfPageBadge {
+
+	private final EmfDiagnosticLevel level;
+	private final Supplier<Integer> countSupplier;
+	private Predicate<IDomRefreshBusEvent> refreshPredicate;
+	private boolean hideZero;
+
+	/**
+	 * Constructs this badge with the given {@link EmfDiagnosticLevel} and count
+	 * supplier.
+	 *
+	 * @param level
+	 *            the {@link EmfDiagnosticLevel} (never <i>null</i>)
+	 * @param countSupplier
+	 *            the {@link Supplier} for the badge counter (never <i>null</i>)
+	 */
+	public EmfPageBadge(EmfDiagnosticLevel level, Supplier<Integer> countSupplier) {
```

Comment:
Hm, maybe we need a title for the badges that pops up when you hover over them?
Currently badges are just arbitrary numbers with different colored backgrounds that dont really tell anything

(Granted, current implementation is self explanatory, but this could be used in more pages in the future)

Reply:
did so
![image](https://user-images.githubusercontent.com/83839745/234010796-42bd9d4f-1cea-40aa-b93e-8e88a7325a0c.png)